components/brave_news/browser/direct_feed_controller.h

```diff
@@ -97,9 +112,24 @@ class DirectFeedController {
                   const GURL& feed_url,
                   const std::unique_ptr<std::string> response_body);
 
+  void FindFeedsImpl(const GURL& possible_feed_or_site_url);
+  void OnFindFeedsImplResponse(
+      const GURL& feed_url,
+      std::vector<mojom::FeedSearchResultItemPtr> results);
+
   raw_ptr<PrefService> prefs_;
   SimpleURLLoaderList url_loaders_;
+
+  // TODO(sko) We should have a way to cancel requests.
+  // e.g. Navigate to different sites, quit app.
+  // Witthout that, some heavy RSS feed parsing work will prevent new feeds from
+  // detection and app from shutdown.
+  std::queue<FindFeedRequest> pending_requests_;
```

Comment:
This seems like it's probably going to be quite important :smile:

Reply:
Yeah, It'd be good to find out how to kill worker thread too.